src/containers/FormikForm/GrepCodesField.tsx

```diff
@@ -6,20 +6,207 @@
+
+const StyledList = styled("ul", {
+  base: { listStyle: "none" },
+});
+
+const StyledComboboxContent = styled(ComboboxContent, {
+  base: {
+    maxHeight: "unset",
+    overflowY: "unset",
+  },
+});
+
+const StyledComboboxList = styled(ComboboxList, {
+  base: {
+    maxHeight: "surface.xsmall",
+    overflowY: "auto",
+  },
+});
+
+export const convertGrepCodesToObject = async (grepCodes: string[]): Promise<Record<string, string>> => {
+  const grepCodesWithTitle = await Promise.all(
+    grepCodes.map(async (c) => {
+      const grepCodeTitle = await fetchGrepCodeTitle(c);
+      return {
+        [c]: grepCodeTitle ? `${c} - ${grepCodeTitle}` : c,
+      };
+    }),
+  );
+  return Object.assign({}, ...grepCodesWithTitle);
+};
 
 const GrepCodesField = () => {
   const { t } = useTranslation();
+  const translations = useComboboxTranslations();
+  const [field, _, helpers] = useField<string[]>("grepCodes");
+  const [grepCodes, setGrepCodes] = useState<Record<string, string>>({});
+
+  const { query, setQuery } = usePaginatedQuery();
+  const searchQuery = useGrepCodesSearch({ input: query });
+
+  useEffect(() => {
+    (async () => {
+      const grepCodesObject = await convertGrepCodesToObject(field.value);
+      setGrepCodes(grepCodesObject);
+    })();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const fetchGrepCodeTitles = async (newGrepCodes: string[]): Promise<GrepCode[]> => {
+    const newGrepCodeNames: GrepCode[] = [];
+    for (const grepCode of newGrepCodes) {
+      const grepCodeTitle = await fetchGrepCodeTitle(grepCode);
+      const isGrepCodeSaved = grepCodes[grepCode];
+
+      if (grepCodeTitle && !isGrepCodeSaved && isGrepCodeValid(grepCode)) {
+        newGrepCodeNames.push({
+          code: grepCode,
+          title: `${grepCode} - ${grepCodeTitle}`,
+        });
+      } else if (!isGrepCodeSaved) {
+        setTimeout(() => {
+          helpers.setError(`${t("errorMessage.grepCodes")}${grepCode}`);
+        }, 0);
+      }
+    }
+    return newGrepCodeNames;
+  };
+
+  const updateGrepCodes = async (values: string[]) => {
+    helpers.setError(undefined);
+    const trimmedValues = uniq(values.map((v) => v.toUpperCase().trim()));
+    // Grep code is added
+    if (trimmedValues.length > field.value.length) {
+      const addedGrepCodes = difference(trimmedValues, field.value);
+      const grepCodesWithNames = await fetchGrepCodeTitles(addedGrepCodes);
+      const grepCodesObject = grepCodesWithNames.reduce((acc, c) => ({ ...acc, [c.code]: c.title }), grepCodes);
+      setGrepCodes(grepCodesObject);
+      helpers.setValue(Object.keys(grepCodesObject));
+      return;
+    }
+    // Grep code is removed
+    if (trimmedValues.length < field.value.length) {
+      const filteredGrepCodes = trimmedValues.reduce(
+        (acc, c) => (values.includes(c) ? { ...acc, [c]: grepCodes[c] } : acc),
+        {},
+      );
```

Comment:
Synes denne var litt vanskelig å lese. Går det an å skrive den på en annen måte?

Reply:
Er den så vanskelig da? kan skrive en kommentar eventuelt? eller så er jeg åpen for innspill på hvordan det kunne blitt gjort annerledes! 😄